plugin/src/main/java/org/opensearch/ml/sdkclient/DDBOpenSearchClient.java

```diff
@@ -38,30 +42,41 @@
 import org.opensearch.sdk.UpdateDataObjectRequest;
 import org.opensearch.sdk.UpdateDataObjectResponse;
 
-import lombok.AllArgsConstructor;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.node.ArrayNode;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import com.google.common.annotations.VisibleForTesting;
+
 import lombok.extern.log4j.Log4j2;
 import software.amazon.awssdk.services.dynamodb.DynamoDbClient;
 import software.amazon.awssdk.services.dynamodb.model.AttributeValue;
 import software.amazon.awssdk.services.dynamodb.model.DeleteItemRequest;
 import software.amazon.awssdk.services.dynamodb.model.GetItemRequest;
 import software.amazon.awssdk.services.dynamodb.model.GetItemResponse;
 import software.amazon.awssdk.services.dynamodb.model.PutItemRequest;
+import software.amazon.awssdk.services.dynamodb.model.UpdateItemRequest;
 
 /**
  * DDB implementation of {@link SdkClient}. DDB table name will be mapped to index name.
  *
  */
-@AllArgsConstructor
 @Log4j2
 public class DDBOpenSearchClient implements SdkClient {
 
+    private static final ObjectMapper OBJECT_MAPPER = new ObjectMapper();
     private static final String DEFAULT_TENANT = "DEFAULT_TENANT";
 
     private static final String HASH_KEY = "tenant_id";
     private static final String RANGE_KEY = "id";
-    private static final String SOURCE = "source";
 
     private DynamoDbClient dynamoDbClient;
+    private RemoteClusterIndicesClient remoteClusterIndicesClient;
+
+    public DDBOpenSearchClient(DynamoDbClient dynamoDbClient, RemoteClusterIndicesClient remoteClusterIndicesClient) {
```

Comment:
Can we add java doc? Seems like `remoteClusterIndicesClient` will only be used for search operation? So if customer doesn't need the search functionality then they can just send null for `remoteClusterIndicesClient`?